packages/builtin/array.pony

```diff
@@ -128,13 +128,19 @@ class Array[A] is Seq[A]
     """
     _size = len
 
-    if alloc > len then
-      _alloc = alloc
+    if ptr.is_null() then
+      _size = 0
+      _alloc = 1
+      _ptr = Pointer[A]._alloc(_alloc)
```

Comment:
I don't think this is quite right, see https://github.com/ponylang/ponyc/blob/e6910e81d2e2ad539f7cbe51924f03f52b3d882b/packages/builtin/array.pony#L120 where zero-length arrays are initialized. This should just get `_alloc = 0` and use the/a null pointer. That way it doesn't need to allocate until an element is actually added.

Did you run into difficulty with that?

Reply:
Ah, I see, you inherited this from the String constructor (and really surprised we didn't notice that one before). The reason String needs an alloc is because it creates a null-terminate string and as such needs a char for that

Reply:
Yes, I saw that the String constructor was handling it correctly, and took the code from there, but you're right that we don't need to allocate for a zero-length array, I'll fix that.

Reply:
@jasoncarr0 This should be fixed now. Does the new version look good to you?